src/test/operators/table_scan_test.cpp

```diff
@@ -754,25 +788,33 @@ TEST_P(OperatorsTableScanTest, GetImpl) {
-  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(TableScan{get_int_float_op(),
-                                                                        in_(column_a, list_(1, 2, 3))}.create_impl().get()));  // NOLINT
-  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(TableScan{get_int_float_op(), and_(greater_than_(column_a, 5),   // NOLINT
-                                                                                             less_than_(column_b,
-                                                                                                        6))}.create_impl().get()));  // NOLINT
-  EXPECT_TRUE(dynamic_cast<ColumnIsNullTableScanImpl*>(TableScan{get_int_float_with_null_op(), is_null_(column_an)}.create_impl().get()));  // NOLINT
-  EXPECT_TRUE(dynamic_cast<ColumnIsNullTableScanImpl*>(TableScan{get_int_float_with_null_op(), is_not_null_(column_an)}.create_impl().get()));  // NOLINT
-  // clang-format on
+  const auto pqp = std::make_shared<Limit>(
+      std::make_shared<Projection>(get_int_string_op(), expression_vector(PQPColumnExpression::from_table(
+                                                            *get_int_float_op()->get_output(), "a"))),
+      to_expression(int64_t{1}));
+
+  EXPECT_TRUE(dynamic_cast<ColumnVsValueTableScanImpl*>(
+      TableScan{get_int_float_op(), equals_(column_a, 5)}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnVsValueTableScanImpl*>(
+      TableScan{get_int_float_op(), equals_(5, column_a)}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnVsValueTableScanImpl*>(
+      TableScan{get_int_float_op(), equals_(5, column_a)}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnVsColumnTableScanImpl*>(
+      TableScan{get_int_float_op(), equals_(column_b, column_a)}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnLikeTableScanImpl*>(
+      TableScan{get_int_string_op(), like_(column_s, "%s%")}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(
+      TableScan{get_int_string_op(), like_("hello", "%s%")}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(
+      TableScan{get_int_float_op(), in_(column_a, list_(1, 2, 3))}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(
+      TableScan{get_int_float_op(), in_(column_a, list_(1, 2, 3))}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ExpressionEvaluatorTableScanImpl*>(
+      TableScan{get_int_float_op(), and_(greater_than_(column_a, 5), less_than_(column_b, 6))}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnIsNullTableScanImpl*>(
+      TableScan{get_int_float_with_null_op(), is_null_(column_an)}.create_impl().get()));
+  EXPECT_TRUE(dynamic_cast<ColumnIsNullTableScanImpl*>(
+      TableScan{get_int_float_with_null_op(), is_not_null_(column_an)}.create_impl().get()));
```

Comment:
Now these tests use "create_impl" to check for the impl type and above you use "impl_description". Can we settle on one, preferably the former (since I do not like deducing types from strings)?

Reply:
The former doesn't work because the replacement happens in the execute method. We can't access `_impl` either, because that gets destroyed by `_on_cleanup`. Unless we want to add an extra enum, I can't think of anything better.

Reply:
But, `table_scan->create_impl()->description()` would always be the same as `table_scan->impl_description()`, no?

Reply:
No, because just calling `table_scan->create_impl()` does not make sure that `_on_execute` was called. That is where we replace the predicate.

Reply:
Why not let `create_impl` rewrite the predicate and leave `TableScan::_predicate` unaltered, thus not breaking visualization and not having to have "impl_description()"?

Reply:
Yepp